fractal/cmd/_task.py

```diff
@@ -42,6 +43,11 @@ async def task_collect_pip(
         attributes["python_version"] = python_version
     if package_extras:
         attributes["package_extras"] = package_extras
+    if pinned_dependency:
+        attributes["pinned_package_versions"] = {
+            _name: _version
+            for _name, _version in (x.split("=") for x in pinned_dependency)
```

Comment:
Can we add a bit of preliminary checks here?

For instance let's make sure that `len(x.split("=")) == 2` for each item of the list. If this is not the case, we should fail in a controlled way.

What I'd like to avoid is that a prompt like
```
fractal task collect fractal-tasks-core --pinned-dependency torch 1.12
```
fails with some unclear error.